src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs

```diff
@@ -243,11 +301,60 @@ public void Pack(string nuspecPath, Func<string, string> nuspecPropertyProvider,
             }
             catch (Exception e)
             {
-                string packageType = packSymbols ? "symbol" : "lib";
+                string packageType = "lib";
+                if (creatingSymbolsPackage)
+                {
+                    packageType = "symbol";
+                }
+                else if (packSymbols)
+                {
+                    packageType = "packed";
+                }
                 Log.LogError($"Error when creating nuget {packageType} package from {nuspecPath}. {e}");
             }
         }
 
+        private Manifest TransformManifestToPackedPackageManifest(Manifest manifest)
```

Comment:
Sure but we only expect that we will have either a symbols or transport package but not both so you can mostly share that and essentially reduce the logic down to "CreateSymbols" and "Prefix" and if there is a "Prefix" then generate a transport package otherwise just a symbols package.